Dang, the CUDA builder is exceeding some kind of resources:
```
2026-02-10T23:55:26.7435795Z gmake[3]: *** [CMakeFiles/pyAMReX_2d.dir/build.make:728: CMakeFiles/pyAMReX_2d.dir/src/Particle/ParticleContainer.cpp.o] Terminated
2026-02-10T23:55:26.7441940Z gmake[3]: *** [CMakeFiles/pyAMReX_1d.dir/build.make:698: CMakeFiles/pyAMReX_1d.dir/src/Particle/ParticleContainer.cpp.o] Terminated
2026-02-10T23:55:26.7443121Z gmake[2]: *** [CMakeFiles/Makefile2:1376: CMakeFiles/pyAMReX_1d.dir/all] Terminated
2026-02-10T23:55:26.7460354Z gmake[1]: *** [CMakeFiles/Makefile2:1649: CMakeFiles/pip_install.dir/rule] Terminated
2026-02-10T23:55:26.7465358Z gmake: *** [Makefile:650: pip_install] Terminated
2026-02-10T23:55:26.8153274Z ##[error]The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled.
2026-02-10T23:55:28.0419398Z ##[error]The operation was canceled.
```

Unlikely disk space, because we clean a lot, but maybe RAM or so. We already run with only `-j 2` although we can use 4 vcores these days... We need to split out `ParticleContainer.cpp` more carefully so it is not such a huge banger TU...

--- a/src/Particle/CMakeLists.txt
+++ b/src/Particle/CMakeLists.txt
@@ -2,6 +2,8 @@ foreach(D IN LISTS AMReX_SPACEDIM)
     target_sources(pyAMReX_${D}d
       PRIVATE
         ParticleContainer.cpp
+        ParticleContainer_SoA.cpp
+        ParticleContainer_tests.cpp
         ParticleContainer_FHDeX.cpp
         ParticleContainer_ImpactX.cpp
         ParticleContainer_WarpX.cpp

--- a/src/Particle/ParticleContainer.cpp
+++ b/src/Particle/ParticleContainer.cpp
@@ -82,6 +82,8 @@ namespace
 }
 
 // forward declarations
+void init_ParticleContainer_soa(py::module& m);
+void init_ParticleContainer_tests(py::module& m);
 void init_ParticleContainer_FHDeX(py::module& m);
 void init_ParticleContainer_ImpactX(py::module& m);
 void init_ParticleContainer_WarpX(py::module& m);
@@ -94,15 +96,12 @@ void init_ParticleContainer(py::module& m) {
 
     // most common case: ND particle + runtime attributes
     //   pure SoA
-    make_ParticleContainer_and_Iterators<
-        SoAParticle<AMREX_SPACEDIM, 0>,
-                    AMREX_SPACEDIM, 0
-    >(m);
+    init_ParticleContainer_soa(m);
     //   legacy AoS + SoA
     //make_ParticleContainer_and_Iterators<Particle<0, 0>, 0, 0>(m);
 
     // used in tests
-    make_ParticleContainer_and_Iterators<Particle<2, 1>, 3, 1>(m);
+    init_ParticleContainer_tests(m);
 
     // application codes
     init_ParticleContainer_FHDeX(m);

--- a/src/Particle/ParticleContainer_SoA.cpp
+++ b/src/Particle/ParticleContainer_SoA.cpp
@@ -1,0 +1,20 @@
+/* Copyright 2022 The AMReX Community
+ *
+ * Authors: Ryan Sandberg, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "ParticleContainer.H"
+
+#include <AMReX_Particle.H>
+
+
+void init_ParticleContainer_soa(py::module& m) {
+    using namespace amrex;
+
+    // most common case: ND particle + runtime attributes
+    //   pure SoA
+    make_ParticleContainer_and_Iterators<
+        SoAParticle<AMREX_SPACEDIM, 0>,
+                    AMREX_SPACEDIM, 0
+    >(m);
+}

--- a/src/Particle/ParticleContainer_tests.cpp
+++ b/src/Particle/ParticleContainer_tests.cpp
@@ -1,0 +1,15 @@
+/* Copyright 2022 The AMReX Community
+ *
+ * Authors: Ryan Sandberg, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "ParticleContainer.H"
+
+#include <AMReX_Particle.H>
+
+
+void init_ParticleContainer_tests(py::module& m) {
+    using namespace amrex;
+
+    make_ParticleContainer_and_Iterators<Particle<2, 1>, 3, 1>(m);
+}

